File: backend/src/main/java/com/microflow/agent/domain/model/AgentDiagnostic.java
package com.microflow.agent.domain.model;

public record AgentDiagnostic(
        String agentKey,
        String provider,
        String endpointUrl,
        boolean enabled,
        boolean credentialConfigured,
        String roleStrategy,
        String status,
        String detail,
        long latencyMillis,
        String checkedAt
) {
}

File: frontend/lib/features/agents/domain/entities/agent_diagnostic.dart
class AgentDiagnostic {
  const AgentDiagnostic({
    required this.agentKey,
    required this.provider,
    required this.endpointUrl,
    required this.enabled,
    required this.credentialConfigured,
    required this.roleStrategy,
    required this.status,
    required this.detail,
    required this.latencyMillis,
    required this.checkedAt,
  });

  final String agentKey;
  final String provider;
  final String endpointUrl;
  final bool enabled;
  final bool credentialConfigured;
  final String roleStrategy;
  final String status;
  final String detail;
  final int latencyMillis;
  final String checkedAt;
}
